src/main/java/edu/cornell/kfs/module/ar/document/web/struts/CuContractsGrantsInvoiceDocumentAction.java

```diff
@@ -55,4 +179,15 @@ private KualiDecimal findAwardBudgetTotal(ContractsGrantsInvoiceDocumentForm con
         return budgetTotalAmount;
     }
 
+    private Pair<Date, Date> parseDateRange(String dateRange) throws ParseException {
+        if (StringUtils.isEmpty(dateRange) || dateRange.length() != 24) {
+            dateRange = dateRange == null ? KFSConstants.EMPTY_STRING : dateRange;
```

Comment:
The lines above could use some of the cleanup mentioned earlier (use a constant for expected length, use a StringUtils helper method for null/blank defaults, etc.).